snuba/datasets/schema.py

```diff
@@ -34,3 +39,81 @@ def get_table_name(self):
         """
         table_name = self.__local_table_name if local_dataset_mode() else self.__dist_table_name
         return self.__make_test_table(table_name)
+
+    def _get_distributed_engine(self):
+        return """Distributed(%(cluster)s, %(database)s, %(local_table)s, %(sharding_key)s);""" % {
+            'cluster': self.__clickhouse_cluster,
+            'database': self.__database,
+            'local_table': self.get_local_table_name(),
+            'sharding_key': self.__sharding_key,
+        }
+
+    def _get_table_definition(self, name, engine):
+        return """
+        CREATE TABLE IF NOT EXISTS %(name)s (%(columns)s) ENGINE = %(engine)s""" % {
+            'columns': self.__all_columns.for_schema(),
+            'engine': engine,
+            'name': name,
+        }
+
+    def get_local_table_definition(self, replicated=False):
+        return self._get_table_definition(
+            self.get_local_table_name(),
+            self._get_local_engine(replicated)
+        )
+
+    def get_dist_table_definition(self):
+        assert self.__clickhouse_cluster, "CLICKHOUSE_CLUSTER is not set."
+
+        return self._get_table_definition(
+            self.get_table_name(replicated=True),
+            self.get_distributed_engine()
+        )
+
+    def _get_local_engine(self, replicated=False):
+        pass
+
+    def get_all_columns(self):
+        return self.__all_columns
+
+
+class ReplacingMergeTreeSchema(TableSchema):
```

Comment:
I think we should revisit this abstraction later when we will have clarity on how we want to create distributed tables in production.
Right now all the methods to create a distribute table are never used. Depending on how we decide to handle this in the long term we may want to remove this class and let the dataset provide the DDL to generate the engine.

Reply:
Makes sense to me.